databricks_cli/unity_catalog/cred_cli.py

```diff
@@ -50,16 +124,42 @@
 # Until that is fixed (should return a 400), show full error trace.
 #@eat_exceptions
 @provide_api_client
-def create_credential_cli(api_client, skip_val, json_file, json):
+def create_credential_cli(api_client, name, aws_iam_role_arn,
+                          az_sp_directory_id, az_sp_application_id, az_sp_client_secret,
+                          az_mi_access_connector_id, az_mi_id, gcp_sak_email,
+                          gcp_sak_private_key_id, gcp_sak_private_key, comment,
+                          skip_val, json_file, json):
     """
     Create new storage credential.
 
     The public specification for the JSON request is in development.
     """
-    json_cli_base(json_file, json,
-                  lambda json: UnityCatalogApi(api_client).create_storage_credential(json,
-                                                                                     skip_val),
-                  encode_utf8=True)
+    has_credential_flag = (
+        (aws_iam_role_arn is not None) or
+        (az_sp_directory_id is not None) or (az_sp_application_id is not None) or
+        (az_sp_client_secret is not None) or (az_mi_access_connector_id is not None) or
+        (az_mi_id is not None) or (gcp_sak_email is not None) or
+        (gcp_sak_private_key_id is not None) or (gcp_sak_private_key is not None))
+    if ((name is not None) or has_credential_flag or (comment is not None)):
+        if (json_file is not None) or (json is not None):
+            raise ValueError('Cannot specify JSON if any other creation flags are specified')
+        data = {
+            'name': name,
+            'comment': comment
+        }
```

Comment:
Looks like we are leaving it completely to the server to validate the input -- e.g., we are not checking that `data` has required fields like `name`.  I suppose that's reasonable (as we also do not validate JSON inputs), but I wanted to confirm that this is our approach for input validation at the CLI.

Reply:
Yes, that's the idea. Instead of replicating the full validation logic, just letting the server do it and then return the error message would make things easier to maintain.

Reply:
Makes sense.